onnx/defs/parser.cc

```diff
@@ -43,6 +43,35 @@ Status OnnxParser::Parse(char open, IdList& idlist, char close) {
   return Status::OK();
 }
 
+Status OnnxParser::Parse(IdList& idlist, AttrList& attrlist) {
+  idlist.Clear();
+  attrlist.Clear();
+  do {
+    std::string id;
+    bool is_assignment = PeekIsIdentifierAssignment(id);
+    if (id.empty())
+      return Status::OK(); // Treat as empty list of identifiers
+    if (is_assignment) {
+      PARSE(*attrlist.Add());
```

Comment:
just wondering: may be the logic could be simplified if we extend the `OnnxParser::Parse(AttributeProto& attr)` method to take the `id` as an optional parameter?

Reply:
OnnxParser::Parse(AttributeProto& attr) is called from: (1) OnnxParser::Parse(NodeProto& node); (2) OnnxParser::Parse(FunctionProto& fn). For the first case, id is not used. If we want to merge the 2 calls, we can in the first case call OnnxParser::Parse(IdList& idlist, AttrList& attrlist) and ignore the idList. I feel that makes the code less readable. Maybe I understand it wrong?